modules/base/rendering/renderablemodel.cpp

```diff
@@ -279,103 +282,173 @@ RenderableModel::RenderableModel(const ghoul::Dictionary& dictionary)
+        }
+        else {
+            _enableAnimation = *p.enableAnimation;
+            _geometry->enableAnimation(_enableAnimation.value());
+        }
+    }
+
+    if (p.animationTimeScale.has_value()) {
+        if (!_geometry->hasAnimation()) {
+            LWARNING("Animation time scale given to model without animation");
+        }
+        else if (std::holds_alternative<float>(*p.animationTimeScale)) {
+            _geometry->setTimeScale(std::get<float>(*p.animationTimeScale));
+        }
+        else if (std::holds_alternative<Parameters::AnimationTimeUnit>(*p.animationTimeScale)) {
+            Parameters::AnimationTimeUnit animationTimeUnit =
+                std::get<Parameters::AnimationTimeUnit>(*p.animationTimeScale);
+            TimeUnit timeUnit;
+
+            switch (animationTimeUnit) {
+                case Parameters::AnimationTimeUnit::Nanosecond:
+                    timeUnit = TimeUnit::Nanosecond;
+                    break;
+                case Parameters::AnimationTimeUnit::Microsecond:
+                    timeUnit = TimeUnit::Microsecond;
+                    break;
+                case Parameters::AnimationTimeUnit::Millisecond:
+                    timeUnit = TimeUnit::Millisecond;
+                    break;
+                case Parameters::AnimationTimeUnit::Second:
+                    timeUnit = TimeUnit::Second;
+                    break;
+                case Parameters::AnimationTimeUnit::Minute:
+                    timeUnit = TimeUnit::Minute;
+                    break;
+                default:
+                    throw ghoul::MissingCaseException();
             }
+
+            _geometry->setTimeScale(convertTime(1.0, timeUnit, TimeUnit::Second));
+        }
+    }
+
+    if (p.animationMode.has_value()) {
+        if (!_geometry->hasAnimation()) {
+            LWARNING("Animation mode given to model without animation");
+        }
+
+        switch (*p.animationMode) {
+            case Parameters::AnimationMode::LoopFromStart:
+                _animationMode = AnimationMode::LoopFromStart;
+                break;
+            case Parameters::AnimationMode::LoopInfinitely:
+                _animationMode = AnimationMode::LoopInfinitely;
+                break;
+            case Parameters::AnimationMode::BounceFromStart:
+                _animationMode = AnimationMode::BounceFromStart;
+                break;
+            case Parameters::AnimationMode::BounceInfinitely:
+                _animationMode = AnimationMode::BounceInfinitely;
+                break;
+            case Parameters::AnimationMode::Once:
+            default:
+                _animationMode = AnimationMode::Once;
+                break;
         }
     }
 
-    if (dictionary.hasKey(ModelTransformInfo.identifier)) {
-        _modelTransform = dictionary.value<glm::dmat3>(ModelTransformInfo.identifier);
+    if (p.modelTransform.has_value()) {
```

Comment:
More concise to write:
`_modelTransform = p.modelTransform.value_or(_modelTransform);`
and the same for the rest here